### PR TITLE
Allow comments near Civet directives

### PR DESCRIPTION
**Pull request by p2edwards**
Preserve comments near Civet directives

xref [Discord discussion](https://discord.com/channels/933472021310996512/933472021310996515/1165013088697860146)

### CODE REVIEW

File: source/parser.hera [hunk 1]
```diff
@@ -6524,8 +6524,12 @@ Shebang
   /#![^\r\n]*/ EOL
 
 CivetPrologue
-  [\t ]* DoubleQuote CivetPrologueContent:content DoubleQuote SimpleStatementDelimiter EOS? -> content
-  [\t ]* SingleQuote CivetPrologueContent:content SingleQuote SimpleStatementDelimiter EOS? -> content
+  [\t ]* DoubleQuote CivetPrologueContent:content DoubleQuote SimpleStatementDelimiter _? EOL? EOS?:eos ->
+    content.children.push(eos)
```
Comment (STRd6) [inline source/parser.hera:6528]: Thanks for the PR! One small issue is that the way caching works in the parser we want to avoid mutating the objects passed to the handler rules.

For this case we could do something like:

```javascript
return {...content, children: [...content.children, eos]}
```

Comment (p2edwards) [reply source/parser.hera]: Thanks! I've incorporated your suggestion verbatim.

File: source/parser.hera [hunk 2]
```diff
@@ -6524,8 +6524,10 @@ Shebang
   /#![^\r\n]*/ EOL
 
 CivetPrologue
-  [\t ]* DoubleQuote CivetPrologueContent:content DoubleQuote SimpleStatementDelimiter EOS? -> content
-  [\t ]* SingleQuote CivetPrologueContent:content SingleQuote SimpleStatementDelimiter EOS? -> content
+  [\t ]* DoubleQuote CivetPrologueContent:content DoubleQuote SimpleStatementDelimiter _? EOL? EOS?:eos ->
```
Comment (edemaine) [inline source/parser.hera:6527]: Why did you add `_? EOL?`? In particular, this will discard inline comments (in `_`), and seems like it can consume two newlines. But maybe I'm missing something?

Comment (p2edwards) [reply source/parser.hera:6527]: Oh interesting! I overlooked that `_` included InlineComment.

With `_? EOL?` I was looking for a simple way to do these 2 things:

1. Trim initial whitespace from an end-of-line comment (or the start of a multiline one)
2. Omit the blank line if there's no comment after a directive

So that

```coffee
"civet -implicitReturns" // hello!
"civet autoLet"
// hi!
```

becomes

```ts
// hello!
// hi!
```

rather than 
```ts
 // hello!

// hi!
```

Comment (edemaine) [reply source/parser.hera]: Ah, I see. Thanks for the explanation! That makes sense now.

The `Init` rule where this gets used has (recently added) support for comments, so I wonder if the following would actually be a simpler solution to this problem:

```
... PrologueEOL [ \t]* ( EOL / &RestOfLine ) -> content
```

This should leave the comments on the remainder of the line for further processing, but only if there's something other than a newline, after trimming spaces/tabs.

**Review by STRd6 — APPROVED** [review]

**Comment by STRd6** [comment]
Thanks!
